tests/network/primary_pod_network.go

```diff
@@ -70,25 +70,69 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 		})
 
 		Context("VMI connected to the pod network using bridge binding", func() {
-			var vmi *v1.VirtualMachineInstance
+			When("Guest Agent exists", func() {
+				var (
+					vmi   *v1.VirtualMachineInstance
+					vmiIP = func() (string, error) {
+						vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+						if err != nil {
+							return "", err
+						}
+						return vmi.Status.Interfaces[0].IP, nil
+					}
 
-			BeforeEach(func() {
-				vmi = setupVMI(virtClient, vmiWithBridgeBinding())
-			})
+					vmiIPs = func() ([]string, error) {
+						vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+						if err != nil {
+							return nil, err
+						}
+						return vmi.Status.Interfaces[0].IPs, nil
+					}
+				)
+				BeforeEach(func() {
+					var err error
 
-			AfterEach(func() {
-				cleanupVMI(virtClient, vmi)
+					vmi, err = newFedoraWithGuestAgentAndDefaultInterface(libvmi.InterfaceDeviceWithBridgeBinding())
+					Expect(err).NotTo(HaveOccurred())
+
+					vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+					Expect(err).NotTo(HaveOccurred())
+					tests.WaitForSuccessfulVMIStart(vmi)
+					tests.WaitAgentConnected(virtClient, vmi)
+				})
+
+				AfterEach(func() {
+					cleanupVMI(virtClient, vmi)
+				})
+
+				It("should report PodIP/s as its own on interface status", func() {
+					vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
+					Eventually(vmiIP).Should(WithTransform(removeCIDR, Equal(vmiPod.Status.PodIP)), "should contain VMI Status IP as Pod status ip")
```

Comment:
removeCIDR doesnt get args?
also you need to use the function that choose the primary ip, as the primary ip might change, and its reflected on PodIP

Reply:
> removeCIDR doesnt get args?

No, it just remove the CIDR or return the ip as it if there is an error removing it (there is no CIDR).

> also you need to use the function that choose the primary ip, as the primary ip might change, and its reflected on PodIP

This is not removing the CIDR from PodIP this is removing the CIDR from vmiIP

Reply:
aint it a function ?
https://github.com/kubevirt/kubevirt/pull/4492/files#diff-de78d9c6855348436f7884b48f68bdf59eb7b35b07de5b89d562b87507625cc2R239

Reply:
> aint it a function with args?
> https://github.com/kubevirt/kubevirt/pull/4492/files#diff-de78d9c6855348436f7884b48f68bdf59eb7b35b07de5b89d562b87507625cc2R239

the Arg is `vmiIP`. `removeCIDR` does the "transform" to `vmiIP` and then match against `Equal(...)`
https://godoc.org/github.com/onsi/gomega#WithTransform